src/managed/Microsoft.Extensions.DependencyModel/StreamBufferWriter.Netstandard2_0.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    internal struct StreamBufferWriter : IBufferWriter<byte>, IDisposable
```

Comment:
We should consider adding such an adapter between IBufferWriter and Stream, publicly.

cc @stephentoub, @davidfowl

Reply:
Yes please. We have something similar in SignalR https://github.com/aspnet/AspNetCore/blob/958e9049894137bb08c0d239dca0cda7c850a691/src/SignalR/common/Shared/MemoryBufferWriter.cs#L14 (ignore the fact that it's also a Stream, we just did that for other reasons).

Reply:
Also the PipeWriter implemenation over a Stream is 90% similar https://github.com/aspnet/AspNetCore/blob/958e9049894137bb08c0d239dca0cda7c850a691/src/SignalR/common/Shared/MemoryBufferWriter.cs#L14.

Reply:
This is something that @JeremyKuhne might have looked into.
/cc @danmosemsft

Reply:
https://github.com/dotnet/corefx/issues/34894